internal/config/settings.go

```
@@ -32,9 +48,17 @@ func NewSettings() *Settings {
 	envSettings := NewENVSettings()
 	flags := NewFlags()
 
+	fileConfig, err := ufiles.ParseJSON[SettingsFromJSON](
+		lo.CoalesceOrEmpty(flags.JSONConfigPath, envSettings.ConfigJSONPath),
+	)
+
+	if err != nil {
```

Существует хороший приннцип "Fail early": если при инициализации программы что-то пошло не так, то лучше не пытаться исправить ошибку, а просто завершить процесс с ненулевым кодом. Так будет сразу видно, что что-то пошло не так.

Если сделать fallback на дефолтные значения, то это может быть неожиданно для клиента: он передавал один параметром, а сервис запустился с другим. Это может привести к неожиданному (а значит непонятному) поведению сервиса.

Исправлять некритические ошибки во время работы сервиса -- хорошо, при инициализации -- скорее плохо